### PR TITLE
[Feature] Implement Text Selection Controller

### PR DESCRIPTION
**Pull request by rizikolik**
This PR implements a new controller named text selection.

## Related tickets

-   [WRS-715](https://support.chili-publish.com/projects/WRS/issues/WRS-715)
- [WRS-715](https://github.com/chili-publish/editor-workspace/pull/384)

### CODE REVIEW

File: src/controllers/TextSelectionController.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { EditorAPI } from '../../types/CommonTypes';
+import { TextStyleUpdateType } from '../../types/TextStyleTypes';
+/**
+ * The TextSelectionController is responsible for all communication regarding text styles.
+ * Methods inside this controller can be called by `window.SDK.textStyle.{method-name}`
+ */
+export class TextSelectionController {
```
Comment (brapoprod) [inline src/controllers/TextSelectionController.ts:7]: this doesnt feel correct, what has text styles to do with text selection? naming feels off

File: src/controllers/TextSelectionController.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * @ignore
+     */
+    constructor(editorAPI: EditorAPI) {
+        this.#editorAPI = editorAPI;
+    }
+
+    /**
+     * This method returns all debug logs
```
Comment (brapoprod) [inline src/controllers/TextSelectionController.ts:21]: debug logs?

File: src/index.ts [hunk 3]
```diff
@@ -108,11 +123,13 @@ export class SDK {
                 onSelectedToolChanged: this.subscriber.onSelectedToolChanged,
                 onUndoStateChanged: this.subscriber.onUndoStateChanged,
                 onSelectedLayoutFramesChanged: this.subscriber.onSelectedLayoutFramesChanged,
+                onSelectedTextStyleChanged: this.subscriber.onSelectedTextStyleChanged,
             },
             this.setConnection,
             this.config.editorId,
         );
         this.editorAPI = connection?.promise.then((editorAPI) => {
+            console.log(editorAPI, 'HEERE WE GOOO');
```
Comment (brapoprod) [inline src/index.ts:132]: wild log appeared

**Review by abdelhalimkhouas — APPROVED** [review]
Maybe you can also rename the file from `TextSelectionController` to `TextStyleController`

**Review by alitunaarel — APPROVED** [review]

File: src/controllers/TextSelectionController.ts (hunk 2 above)
Comment (rizikolik) [reply src/controllers/TextSelectionController.ts]: Done 3b231437acc17ad1deb4db49d47128d9ebdca9a9

File: src/index.ts (hunk 3 above)
Comment (rizikolik) [reply src/index.ts:132]: Done 3b231437acc17ad1deb4db49d47128d9ebdca9a9

File: src/controllers/TextSelectionController.ts (hunk 1 above)
Comment (rizikolik) [reply src/controllers/TextSelectionController.ts]: DDone 3b231437acc17ad1deb4db49d47128d9ebdca9a9

File: src/index.ts (hunk 3 above)
Comment (brapoprod) [reply src/index.ts:132]: can you remove it?

Comment (rizikolik) [reply src/index.ts]: Sorry,done 86f08e5b307652d6007f50c05e316e9044767dc0

**Review by alexandraFlavia9 — APPROVED** [review]

**Review by brapoprod — APPROVED** [review]

**Pull request merged**